addons/mail/static/tests/core/common/store_service.test.js

```diff
@@ -52,3 +52,27 @@ test("store.insert deletes record after relation created it", async () => {
     await assertSteps(["new-1"]);
     expect(store.Message.get({ id: 1 })?.id).toBe(undefined);
 });
+
+test("store.insert different PY model having same JS model", async () => {
+    await start();
+    const store = getService("mail.store");
+    const data = {
+        "discuss.channel": [
+            { id: 1, name: "General" },
+            { id: 2, name: "Sales" },
+        ],
+        "mail.thread": [
+            {
+                id: 1,
+                model: "discuss.channel",
+            },
+        ],
```

Comment:
Note: this test setup doesn't cover the potential case where the first inserted model works and the 2nd one ignored.
It would be best to have a common thread, but also one only in channel, and one only in thread.